Assets/HoloToolkit/Input/Scripts/InputSources/BaseInputSource.cs

```diff
@@ -11,88 +11,11 @@ namespace HoloToolkit.Unity.InputModule
     /// </summary>
     public abstract class BaseInputSource : MonoBehaviour, IInputSource
     {
-        public event EventHandler<InputSourceEventArgs> SourceUp;
```

Comment:
I am for making this change because of all of its benefits, but just pointing out that the reason these event handlers originally existed was to allow developers to hook into the input sources directly if they wanted to bypass the input manager for some reason.

I don't think that this change prevents users from doing that: they could simply use InputManager and have a class that registers itself as the one modal listener for their application, which would achieve the same purpose. Just adding this comment for awareness :)